app/src/main/kotlin/compiler/ast/Program.kt

```diff
@@ -3,6 +3,8 @@ package compiler.ast
 import compiler.input.LocationRange
 
 data class Program(val globals: List<Global>) {
+    var staticFunctionDepth: Int = 0
```

Comment:
This would be the only mutable variable in AST, which would be quite odd. The maximum depth is only used by `DisplayStorage`, so it may as well be calculated there. Alternatively, `DisplayStorage` may receive additional constructor argument with the maximum depth calculated somewhere else.

Reply:
Writing another ast traversal would be redundant. I'm gonna make name resolution return depth as part of the result and include it in `ProgramProperties`. We can then pass depth to `DisplayStorage` constructor.